src/outgoing.rs

```diff
@@ -3,6 +3,8 @@ use super::*;
 #[derive(Debug, PartialEq)]
 pub(crate) enum Outgoing {
   Amount(Amount),
+  All, // send all cardinals
+  Max, // only send cardinals that can pay for their own fees to maximize the output amount
```

Comment:
I don't see a difference between how these two cases are handled in the logic.

Just call it AllCardinal?